python_modules/dagster/dagster/_core/definitions/assets/graph/base_asset_graph.py

```diff
@@ -640,11 +640,19 @@ def get_materializable_roots(self, asset_key: AssetKey) -> AbstractSet[AssetKey]
             and not self.has_materializable_parents(key)
         }
 
-    def validate_partition_mappings(self):
+    def validate_partitions(self):
         for node in self.asset_nodes:
             if node.is_external:
                 continue
 
+            if node.partitions_def:
+                try:
+                    node.partitions_def.validate_partition_definition()
+                except Exception as e:
+                    raise DagsterInvalidDefinitionError(
+                        f"Invalid partition definition for {node.key.to_user_string()}"
+                    ) from e
```

Comment:
I don't think we need to do this wrapping if we just have all of the validation functions raise errors